cmake/modules/kconfig.cmake

```diff
@@ -248,6 +248,7 @@ set(
   ${CONF_FILE_AS_LIST}
   ${shield_conf_files}
   ${OVERLAY_CONFIG_AS_LIST}
+  ${snippet_conf_files}
```

Comment:
kconfig.cmake is used not only by Zephyr build but also sysbuild, so having snippet specifics in here is not nice.
Kconfig relies on CONF_FILE and OVERLAY_CONFIG, and those are fetched by `zephyr_get()`, a call which controls the order of precedence for various scopes, such as sysbuild defined, cache, local, env, in a central place.

Thus, as mentioned here: https://github.com/zephyrproject-rtos/zephyr/pull/57142#discussion_r1176385878
we should instead get the snippets scope into the `zephyr_get()` instead.

I promised to create this work as followup on https://github.com/zephyrproject-rtos/zephyr/pull/56801, which got merged two days ago.
So just have a little patience 😉

Reply:
FYI: #57661